Civi/Token/TokenCompatSubscriber.php

```diff
@@ -131,8 +131,16 @@ public function onRender(TokenRenderEvent $e) {
 
     if ($useSmarty) {
       $smarty = \CRM_Core_Smarty::singleton();
-      $e->string = $smarty->fetch("string:" . $e->string);
+      set_error_handler([$this, 'handleSmartyError'], E_USER_ERROR);
+      $e->string = $smarty->fetch('string:' . $e->string);
+      restore_error_handler();
     }
   }
 
+  public function handleSmartyError($errno, $errstr, $errfile, $errline) {
```

Comment:
Just wondering if this should be in a more generic spot or something given I'm guessing we may want to wrap other places e.g. https://github.com/civicrm/civicrm-core/blob/master/CRM/Mailing/BAO/Mailing.php#L1172 https://github.com/civicrm/civicrm-core/blob/master/CRM/Mailing/BAO/Mailing.php#L1183 https://github.com/civicrm/civicrm-core/blob/a9669d7636857c91cdfa07cc302c9868be648108/CRM/Utils/Token.php#L830 with something like this as well

Reply:
@seamuslee001 those mailing calls probably should call the token processor (unless they are in code which is to be replaced by flexmailer)